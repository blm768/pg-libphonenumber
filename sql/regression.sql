CREATE EXTENSION IF NOT EXISTS pg_libphonenumber;

-- TODO: test in/out functions.

-- Test phone number parsing
select parse_phone_number('555-555-5555', 'US');
select parse_phone_number('005-555-5555', 'US');
select parse_phone_number('555-555-5555555555', 'US');
-- Produces an error from libphonenumber
select parse_phone_number('555-555-55555555555', 'US');

-- Test operators
select parse_phone_number('555-555-5555', 'US') = parse_phone_number('555-555-5555', 'US');
select parse_phone_number('555-555-5555', 'US') <> parse_phone_number('555-055-5555', 'US');
select parse_phone_number('555-555-5555', 'US') = parse_phone_number('555-555-555', 'US');

select parse_phone_number('555-545-5555', 'US') < parse_phone_number('555-555-5555', 'US');
select parse_phone_number('555-565-5555', 'US') > parse_phone_number('555-555-5555', 'US');
select parse_phone_number('555-555-555', 'US') < parse_phone_number('555-555-5555', 'US');
select parse_phone_number('555-555-555', 'US') > parse_phone_number('555-555-5555', 'US');

-- TODO: test country codes.

-- Test packed phone number parsing
select parse_packed_phone_number('555-555-5555', 'US');
select parse_packed_phone_number('005-555-5555', 'US');
-- These two should produce errors because the number is too long.
-- Produces an error in pg-libphonenumber's code
select parse_packed_phone_number('555-555-5555555555', 'US');
-- Produces an error from libphonenumber
select parse_packed_phone_number('555-555-55555555555', 'US');

-- Do we get correct country codes?
-- TODO: expand.
select phone_number_country_code(parse_phone_number('+1-555-555-5555', 'USA'));
select phone_number_country_code(parse_packed_phone_number('+1-555-555-5555', 'USA'));
