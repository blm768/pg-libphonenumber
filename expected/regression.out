CREATE EXTENSION IF NOT EXISTS pg_libphonenumber;
-- TODO: test in/out functions.
-- Test phone number parsing
select parse_phone_number('555-555-5555', 'US');
 parse_phone_number 
--------------------
 +1 555-555-5555
(1 row)

select parse_phone_number('005-555-5555', 'US');
 parse_phone_number 
--------------------
 +1 005-555-5555
(1 row)

select parse_phone_number('555-555-5555555555', 'US');
 parse_phone_number  
---------------------
 +1 5555555555555555
(1 row)

-- Produces an error from libphonenumber
select parse_phone_number('555-555-55555555555', 'US');
ERROR:  unable to parse '555-555-55555555555' as a phone number
DETAIL:  National number is too long
-- Test operators
select parse_phone_number('555-555-5555', 'US') = parse_phone_number('555-555-5555', 'US');
 ?column? 
----------
 t
(1 row)

select parse_phone_number('555-555-5555', 'US') <> parse_phone_number('555-055-5555', 'US');
 ?column? 
----------
 t
(1 row)

select parse_phone_number('555-555-5555', 'US') = parse_phone_number('555-555-555', 'US');
 ?column? 
----------
 f
(1 row)

-- TODO: test country codes.
-- Test packed phone number parsing
select parse_packed_phone_number('555-555-5555', 'US');
 parse_packed_phone_number 
---------------------------
 +1 555-555-5555
(1 row)

select parse_packed_phone_number('005-555-5555', 'US');
 parse_packed_phone_number 
---------------------------
 +1 005-555-5555
(1 row)

-- These two should produce errors because the number is too long.
-- Produces an error in pg-libphonenumber's code
select parse_packed_phone_number('555-555-5555555555', 'US');
ERROR:  phone number '+1 5555555555555555' is too long
DETAIL:  National number is too long
-- Produces an error from libphonenumber
select parse_packed_phone_number('555-555-55555555555', 'US');
ERROR:  unable to parse '555-555-55555555555' as a phone number
DETAIL:  National number is too long
-- Do we get correct country codes?
-- TODO: expand.
select phone_number_country_code(parse_packed_phone_number('+1-555-555-5555', 'USA'));
 phone_number_country_code 
---------------------------
                         1
(1 row)

